File: be/routes/userRoutes.js
const express = require('express');
const router = express.Router();
const {
  getUsers,
  getUserById,
  createUser,
  loginUser,
  registerUser,
  verifyEmail,
  forgotPassword,
  resetPassword,
  checkUser
} = require('../controllers/userController');

router.route('/').get(getUsers).post(createUser);
router.route('/login').put(loginUser);
router.route('/register').put(registerUser);    
router.route('/verify-email').put(verifyEmail);
router.route('/:id').get(getUserById);
router.post('/forgot-password', forgotPassword);
router.post('/reset-password', resetPassword);
router.route('/check-user/:username').get(checkUser);

module.exports = router; 

File: go/routes/user_routes.go
package routes

import (
	"github.com/gin-gonic/gin"
	"spellsbee/controllers"
)

// SetupUserRouter attaches user-related routes to the given router
func SetupUserRouter(router *gin.Engine) {
	api := router.Group("/api")
	{
		api.POST("/register", controllers.RegisterUser)
		api.POST("/login", controllers.LoginUser)
		api.PUT("/verify-email", controllers.VerifyEmail)
		api.POST("/forgot-password", controllers.ForgotPassword)
		api.POST("/reset-password", controllers.ResetPassword)

		api.GET("/users", controllers.GetUsers)
		api.POST("/users", controllers.CreateUser)
		api.GET("/users/:id", controllers.GetUserByID)
		api.GET("/users/check-user/:username", controllers.CheckUser)
	}
}
 